webgui/webapp/src/app/services/aws.service.ts

```diff
@@ -0,0 +1,174 @@
+// src/app/services/aws-pricing.service.ts
+import { Injectable } from '@angular/core';
+import * as AWS from 'aws-sdk';
+import { forkJoin, from, map, Observable } from 'rxjs';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AwsService {
+  private pricing: AWS.Pricing;
+
+  constructor() {
+    // Initialize AWS Pricing API
+    // Pricing API is only available in 'us-east-1' region
+    this.pricing = new AWS.Pricing({
+      region: 'us-east-1',

```

Comment:
Kita set sebagai default aja yang ini
`region: 'ap-southeast-3'`

Reply:
ok

Reply:
eh wait, ini defaultnya us-east-1 kalo yg southeast gadapet mas, jadi filternya disini

 const params: AWS.Pricing.GetProductsRequest = {
      MaxResults: 10,
      ServiceCode: 'AmazonS3',
      Filters: [
        { Type: 'TERM_MATCH', Field: 'productFamily', Value: 'Storage' },
        {
          Type: 'TERM_MATCH',
          Field: 'location',
          Value: 'Asia Pacific (Jakarta)',
        },
        { Type: 'TERM_MATCH', Field: 'volumeType', Value: 'Standard' },
        { Type: 'TERM_MATCH', Field: 'storageClass', Value: 'General Purpose' },
      ],
    };

Reply:
bisa gw ganti regionCode ap-southeast-3

Reply:
ok seep2... good to know... karena beda region, akan selisih $ 0.00x usd tergantung dari resources-nya